### PR TITLE
Initial zoom in Zoomer is incorrect

### PR DESCRIPTION
**Pull request by ToniMarc1990**
+ Clamp the plot coordinates to the axis range before querying the values for display
+ Solves issue #636

### CODE REVIEW

File: chartfx-chart/src/main/java/io/fair_acc/chartfx/plugins/Zoomer.java [hunk 1]
```diff
@@ -794,11 +794,11 @@ private Map<Axis, ZoomState> getZoomDataWindows() {
             double dataMin;
             double dataMax;
             if (axis.getSide().isVertical()) {
-                dataMin = axis.getValueForDisplay(minPlotCoordinate.getY());
-                dataMax = axis.getValueForDisplay(maxPlotCoordinate.getY());
+                dataMin = axis.getValueForDisplay(Math.min(axis.getHeight(), minPlotCoordinate.getY()));
+                dataMax = axis.getValueForDisplay(Math.min(axis.getHeight(), maxPlotCoordinate.getY()));
```
Comment (wirew0rm) [inline chartfx-chart/src/main/java/io/fair_acc/chartfx/plugins/Zoomer.java:798]:
```suggestion
                dataMax = axis.getValueForDisplay(Math.max(0, Math.min(axis.getHeight(), maxPlotCoordinate.getY())));
```

Is this sufficient or should it also clamp to `>0`? That would probably the rarer case, but I don't see any fundamental reason for only limiting one direction.

**Review by wirew0rm — COMMENTED** [review]
Hi, welcome to chart-fx and thanks for following this up and providing a fix. I've only looked at the diff yet, but from that it looks good.

File: chartfx-chart/src/main/java/io/fair_acc/chartfx/plugins/Zoomer.java (hunk 1 above)
Comment (ToniMarc1990) [reply chartfx-chart/src/main/java/io/fair_acc/chartfx/plugins/Zoomer.java]: I thought about this case too and would say that it either never occurs or is rare.
I committed you suggested change :-).

**Review by wirew0rm — APPROVED** [review]
LGTM, thanks again for contributing.

Since #636 describes multiple problems, does this fix only one aspects or are they all caused by this root cause? Otherwise i'll investigate and update the issue.

**Comment by ToniMarc1990** [comment]
> LGTM, thanks again for contributing.
> 
> Since #636 describes multiple problems, does this fix only one aspects or are they all caused by this root cause? Otherwise i'll investigate and update the issue.

@wirew0rm: Thanks for merging this fix. When is the next planned/unplanned release? We want to integrate ChartFX in our application and thus need this fix ASAP.

**Comment by RalphSteinhagen** [comment]
@ToniMarc1990 we certainly can target a release soon.

**May I ask you for a favour:**
You seem to indicate that you are using ChartFX in an industrial setting which we are happy to support. We are continuously looking into fostering a more robust community around ChartFX without intruding on user privacy or application purposes.

To highlight ChartFX's versatility and advocate for continuing public support, we would like to gather user stories, applications, and companies without implying any endorsement or financial commitment from said author/company/team.

Would you/your company be willing to contribute by sharing:
  * Your company name, logo and domain of operation (robotics, medical, finance, research, ...).
  * A brief description of how you use ChartFX.
  * Screenshots showing ChartFX in your projects (sans sensitive info).
  * Any public links or materials featuring ChartFX.

Your input will be invaluable in showcasing real-world applications and securing resources for ChartFX's growth. The primary purpose is to highlight the many diverse applications of ChartFX and the JavaFX eco-system in general and to demonstrate how we use [public resources](https://publiccode.eu/en/) to benefit not only GSI/FAIR but also industry and academia alike.

**There are many nay-sayers who claim that engaging with the public costs more than it benefits -- which we'd like to dispel.**

We are aware that being the first in any list is always the hardest. Thus, we would first collect public support from about a dozen companies/institutes before setting up a sub-page. Please reach out on our [matrix channel](https://matrix.to/#/#fair-acc_chart:gitter.im) or via [fair-c2-admin@gsi.de](mailto:fair-c2-admin@gsi.de) if you prefer. 

*N.B. The matrix channel is also a good place to get in contact with the other ChartFX developers in a more timely and inclusive manner.*

**Comment by ToniMarc1990** [comment]
> @ToniMarc1990 we certainly can target a release soon.
> 
> **May I ask you for a favour:** You seem to indicate that you are using ChartFX in an industrial setting which we are happy to support. We are continuously looking into fostering a more robust community around ChartFX without intruding on user privacy or application purposes.
> 
> To highlight ChartFX's versatility and advocate for continuing public support, we would like to gather user stories, applications, and companies without implying any endorsement or financial commitment from said author/company/team.
> 
> Would you/your company be willing to contribute by sharing:
> 
> * Your company name, logo and domain of operation (robotics, medical, finance, research, ...).
> * A brief description of how you use ChartFX.
> * Screenshots showing ChartFX in your projects (sans sensitive info).
> * Any public links or materials featuring ChartFX.
> 
> Your input will be invaluable in showcasing real-world applications and securing resources for ChartFX's growth. The primary purpose is to highlight the many diverse applications of ChartFX and the JavaFX eco-system in general and to demonstrate how we use [public resources](https://publiccode.eu/en/) to benefit not only GSI/FAIR but also industry and academia alike.
> 
> **There are many nay-sayers who claim that engaging with the public costs more than it benefits -- which we'd like to dispel.**
> 
> We are aware that being the first in any list is always the hardest. Thus, we would first collect public support from about a dozen companies/institutes before setting up a sub-page. Please reach out on our [matrix channel](https://matrix.to/#/#fair-acc_chart:gitter.im) or via [fair-c2-admin@gsi.de](mailto:fair-c2-admin@gsi.de) if you prefer.
> 
> _N.B. The matrix channel is also a good place to get in contact with the other ChartFX developers in a more timely and inclusive manner._

@RalphSteinhagen : We are interested in sharing information about us and the usage of ChartFX in our product. I forwarded the inquiry to my manager.

**Comment by ichaus-we** [comment]
> Since #636 describes multiple problems, does this fix only one aspects or are they all caused by this root cause? Otherwise i'll investigate and update the issue.

Thanks for the new release.
Problem 1 (of 4) is fixed by this. See my update in #636

**Comment by RalphSteinhagen** [comment]
@ichaus-we have the same request: You seem to be using ChartFX in an industrial context. Would your company be willing to share details about its domain, example applications/screenshots, and be mentioned (Logo, Name) as one of the libs users?

This would help us a lot with building a strong(er) user base and long-term maintenance commitment.

Thanks in advance for your consideration.

**Pull request merged**